### ISSUE TITLE
User model overridding not working in EVENT_AFTER_LOGIN event handler

### ISSUE THREAD

**Issue by tekord** [2amigos/yii2-usuario#62]
### What steps will reproduce the problem?

Override User model as descibed in doc:

```
'container' => [
    'definitions' => [
        Da\User\Model\User::class => common\models\User::class,
    ],
],
```

Write event handler:

```
Event::on(\yii\web\User::class, \yii\web\User::EVENT_AFTER_LOGIN, function (UserEvent $event) {
			/** @var User $identity */
			$identity = $event->identity;

			/** @var Client $client */
			$client = $identity->someRelation; // FAILED

			...
		});
```
### What is the expected result?

Working program.

### What do you get instead?

I got error:

> Getting unknown property: Da\User\Model\User::someRelation

**Comment by tekord** [comment]
Looks like overriding through the container definition is not working properly. I used classMap earlier, it worked. But not I got error in my controller:
```
    public function beforeAction($action) {
        /** @var MyRelation $relation */
        $relation= $this->user->identity->relation;
```
Getting unknown property: Da\User\Model\User::client

**Comment by tonydspaniard** [comment]
@tekord thanks for your feedback. Is your User model extending from `Da\User\Model\User`? In order to override, we must always extend from that class. Is that the case on this bug?

**Comment by tekord** [comment]
@tonydspaniard yes, extended from `Da\User\Model\User`

**Comment by tonydspaniard** [comment]
@tekord have you tried => `Yii::$app->user->identity->relation`?

**Comment by tonydspaniard** [comment]
I have just reviewed your code and there is no `UserEvent` but a `FormEvent` and is related to `SecurityController`, that is: 

```php 
Event::on(SecurityController:EVENT_AFTER_LOGIN, FormEvent::EVENT_AFTER_LOGIN, function(FormEvent $e) {
   $relation = $user = $e->getForm()->getUser()->relationName;
}); 
```

I think you are on the wrong repository. That way of setting up things seems that you are looking for dektrium's Yii2-user instead.

**Comment by tekord** [comment]
@tonydspaniard `UserEvent` is `yii\web\UserEvent`, I subscribed to core component's event, not dektrium's.

**Comment by tonydspaniard** [comment]
@tekord UserEvent::EVENT_AFTER_LOGIN doesn't exists: https://github.com/2amigos/yii2-usuario/blob/master/src/User/Event/UserEvent.php

**Comment by tekord** [comment]
@tonydspaniard I subscribed to not usario's event:

```
Event::on(\yii\web\User::class, \yii\web\User::EVENT_AFTER_LOGIN, function (UserEvent $event)
```
This event is provided by Yii. I guess this code must work no matter if usario is used or not.

**Comment by tonydspaniard** [comment]
@tekord oh... sorry, now I understood. Apologies, my bad. Unfortunately, that event is not being fired and yes, you are right about it if that event exists on the User model. Is indeed an issue even though the module provides its very own way.

**Comment by tonydspaniard** [comment]
@tekord the problem is on here, see `yii\web\User` code: 

```php 
$this->trigger(self::EVENT_AFTER_LOGIN, new UserEvent([
            'identity' => $identity,
            'cookieBased' => $cookieBased,
            'duration' => $duration,
        ]));
```

That means, that you need to use the Identity class: 

```php 
Event::on(common\models\User::class, \yii\web\User::EVENT_AFTER_LOGIN, function (UserEvent $event)
```

Can you try the above?

**Comment by tonydspaniard** [comment]
@tekord Is indeed very strange as we do call the `afterLogin` of that class. I'll review it and test it as soon I have some free time. Thanks for the feedback.

**Comment by tonydspaniard** [comment]
@tekord I have investigated further the issue and I was right. If you look at the `Bootstrap.php` you'll see that we override the `yii\web\User` (https://github.com/2amigos/yii2-usuario/blob/master/src/User/Bootstrap.php#L146-L152) by using our own `Da\User\Model\User::class`. So, when the `Application` component calls `getUser()` is calling in fact, our configuration, using the `IdentityInterface` we set on our bootstrap class, that is, our User model class and not `yii\web\User`. 

So, we never reach the `beforeLogin` event of the `yii\web\User` as we are not using `yii\web\User` but our own class. The only event possible is the ones described on our docs: `Event::on(SecurityController::class, FormEvent::EVENT_BEFORE_LOGIN, $formEvent)...`

Thank you very much for your feedback. I believe this was important and it was good to refresh my memory. I think I have to add it to the docs.
